File: javascript/word-count/word-count.js
export default class Words {
  count(words) {
    const obj = {}
    words
      .toLowerCase()
      .trim()
      .replace(/[\t\n]+/g, ' ')
      .split(' ')
      .filter(x => x !== '')
      .forEach(word => {
        if (word == 'constructor' || !obj[word]) {
          obj[word] = 1
        } else {
          obj[word] += 1
        }
      })
    return obj
  }
}

File: ruby/word-count/word_count.rb
class Phrase
  def initialize(phrase)
    @phrase = phrase.downcase.gsub(/\b[^a-z0-9']+/, ' ')
  end
  def word_count
    count = {}
    words = @phrase.split(' ').each { |word|
      if (word[0] == "'") then
        word[0] = ''
        word[-1] = ''
      end
      count[word].nil? ? count[word] = 1 : count[word] += 1
    }
    count
  end
end